docs/js-api-reference.rst

```diff
@@ -0,0 +1,57 @@
+JavaScript API Reference
+========================
+
+Following are the functions that are defined in `rtd_sphinx_search.js`_,
+You can overwrite any of them, if any kind of customizations are required.
+
+.. js:autofunction:: debounce
+.. js:autofunction:: convertObjToUrlParams
+.. js:autofunction:: updateUrl
+.. js:autofunction:: createDomNode
+.. js:autofunction:: _is_string
+.. js:autofunction:: _is_array
+.. js:autofunction:: get_section_html
+.. js:autofunction:: getHighlightListData
+.. js:autofunction:: get_domain_html
+.. js:autofunction:: generateSingleResult
+.. js:autofunction:: generateSuggestionsList
+.. js:autofunction:: removeAllActive
+.. js:autofunction:: addActive
+.. js:autofunction:: getInputField
+.. js:autofunction:: removeResults
+.. js:autofunction:: getErrorDiv
+.. js:autofunction:: fetchAndGenerateResults
+.. js:autofunction:: generateAndReturnInitialHtml
+.. js:autofunction:: showSearchModal
+.. js:autofunction:: removeSearchModal
+
+Example: Overwriting A JavaScript Function
+------------------------------------------
+
```

Comment:
Let's remove this section for now, ship this PR, and then do namespacing in another PR.